### ISSUE TITLE
(Question) Why not define `result` less generically?

### ISSUE THREAD

**Issue by itsrainy** [colemickens/twitlistauth#1]
Referencing these lines: https://github.com/colemickens/twitlistauth/blob/master/main.go#L108-L136

Is there a reason you made `result` a `map[string]interface{}` rather than something like the following:

```
type tweetResult struct {
    Id    float64 `json:"id"`
    Users []struct {
        Id float64 `json:"id"`
    } `json:"users"`
}
```

Seems like this way would require a lot fewer type assertions, but I'm not sure if there are downsides I'm missing.

**Comment by colemickens** [comment]
No downsides that I know of. If anything, if it can decode straight into the struct, it might be less runtime type assertions since they're kind of all over the place as you and Andrew noticed.

'Is there a reason' -> laziness. I'll probably do exactly this.

**Comment by itsrainy** [comment]
Right on. Thanks for sharing. It's an interesting way of selectively giving people access to stuff!

**Comment by colemickens** [comment]
Yeah, if you're interested, I actually wrote [the Facebook version first](https://github.com/colemickens/fbrp), and got it working, before realizing they don't let apps access the "user_groups" scope, so I can't actually check if a given user is a member of my Secret Group.

Thus, Twitter Lists.

**Comment by colemickens** [comment]
Also, reopening to keep track of this.
